### ISSUE TITLE
Using DifferentiationInterface?

### ISSUE THREAD

**Issue by gdalle** [jbrea/FitPopulations.jl#5]
Hi, congrats on the new package! I was wondering whether DifferentiationInterface would make your life easier for gradient and hessian computations?

**Comment by epfl-bio322** [comment]
I actually wanted to switch to DifferentiationInterface, but realized it would take me a bit more time to refactor... If I remember correctly, one of the reasons was [this hack](https://github.com/jbrea/LaplacianExpectationMaximization.jl/blob/8c0a0e4bc57725685085606e4568ed2d433be47c/src/diff.jl#L48), where I convert structs such that they work with ForwardDiff. In DifferentiationInterface I could also do this with a closure, but for Enzyme it would make more sense to use a `Cache` and I didn't know how to use a `Cache` with ForwardDiff and this hack... Ideas are welcome :)

**Comment by jbrea** [comment]
I actually wanted to switch to DifferentiationInterface, but realized it would take me a bit more time to refactor... If I remember correctly, one of the reasons was [this hack](https://github.com/jbrea/LaplacianExpectationMaximization.jl/blob/8c0a0e4bc57725685085606e4568ed2d433be47c/src/diff.jl#L48), where I convert structs such that they work with ForwardDiff. In DifferentiationInterface I could also do this with a closure, but for Enzyme it would make more sense to use a `Cache` and I didn't know how to use a `Cache` with ForwardDiff and this hack... Ideas are welcome :)

**Comment by gdalle** [comment]
Yeah at the moment caches only work with arrays, otherwise I'd have to implement a kind of recursive struct explorer and I'd basically be recoding Functors.jl. Sorry I don't have better options at the moment!
